### PR TITLE
allow flexible mini_batch_length

### PR DESCRIPTION
**Pull request by hnyu**
For one_time replayer, sometimes it might be difficult to require that mini_batch_length is divisible by experience length. It will be more flexible if we can remove the check.

Also fix total_size() for the one-time exp replayer. Before it returns the batch size. 

More docstring format corrections.

### CODE REVIEW

File: alf/algorithms/sac_algorithm.py [hunk 1]
```diff
@@ -425,7 +426,7 @@ def _calc_critic_loss(self, training_info):
             value=critic_info.critic2,
             target_value=target_critic)
 
-        critic_loss = critic_loss1.loss + critic_loss2.loss
+        critic_loss = (critic_loss1.loss + critic_loss2.loss) / 2.
```
Comment (Haichao-Zhang) [inline alf/algorithms/sac_algorithm.py]: ``` / 2``` should be removed as each critic network is trained using its original loss instead of a scaled loss.

**Review by Haichao-Zhang — CHANGES_REQUESTED** [review]

File: alf/algorithms/sac_algorithm.py (hunk 1 above)
Comment (hnyu) [reply alf/algorithms/sac_algorithm.py:429]: The absolute scale is not important. We can also argument MSELoss * 0.5 is the way to go. More importantly, the current displayed critic_loss in TB is always doubled, which is confusing.

Comment (hnyu) [reply alf/algorithms/sac_algorithm.py]: Changed to only scaling the extra field for tensorboard display

**Review by Haichao-Zhang — DISMISSED** [review]

File: alf/algorithms/off_policy_algorithm.py [hunk 2]
```diff
@@ -150,9 +150,13 @@ def _train_experience(self, experience, num_updates, mini_batch_size,
...
-        assert length % mini_batch_length == 0, (
-            "length=%s not a multiple of mini_batch_length=%s" %
-            (length, mini_batch_length))
+        if length % mini_batch_length:
+            common.warning_once(
+                "length=%s not a multiple of mini_batch_length=%s" %
+                (length, mini_batch_length))
+            length = length // mini_batch_length * mini_batch_length
+            common.warning_once(
+                "Experience length has been cut to %s" % length)
```
Comment (emailweixu) [inline alf/algorithms/off_policy_algorithm.py:161]: Where is the logic for cutting experience?

Comment (hnyu) [reply alf/algorithms/off_policy_algorithm.py:161]: Somehow missed in this PR. Will update

**Review by emailweixu — APPROVED** [review]
